### PR TITLE
Add instances for Data.Sequence

### LINKED ISSUE #2: Add instances for Data.Sequence
I believe `Seq` should be both `Unfoldable` and `Unfolder`.

```haskell
instance Unfoldable S.Seq where
  unfold fa = choose [ pure empty, (S.<|) <$> fa <*> unfold fa ]

instance Unfolder S.Seq where
  choose = foldl' (<|>) empty
  chooseInt n = S.fromFunction n id
```

### PR DESCRIPTION
**Pull request by treeowl**
* Make `Data.Sequence.Seq` an instance of `Unfoldable` and
of `Unfolder`.

* Add a `.gitignore` file.

* Add `Other-extensions` field to `.cabal`.

Fixes #2

### CODE REVIEW

File: src/Data/Unfoldable.hs [hunk 1]
```diff
@@ -230,4 +231,9 @@ instance (Unfoldable p, Unfoldable q) => Unfoldable (Compose p q) where
...
 instance Unfoldable f => Unfoldable (Reverse f) where
   unfold fa = choose
-    [ Reverse <$> getDualA (unfold (DualA fa)) ]
\ No newline at end of file
+    [ Reverse <$> getDualA (unfold (DualA fa)) ]
+
+instance Unfoldable S.Seq where
+  unfold fa = choose
+    [ pure empty
+    , (S.<|) <$> fa <*> unfold fa ]
```
Comment (sjoerdvisscher) [inline src/Data/Unfoldable.hs]: Would it make sense to also add `(S.|>) <$> unfold fa <*> fa`?

**Comment by treeowl** [comment]
That sounds likely, but I don't really understand Unfoldable yet.

**Comment by treeowl** [comment]
Oh, but we have to pick just one, right? Seq generally behaves like []
unless there's a good reason to behave otherwise.

**Comment by sjoerdvisscher** [comment]
Hmm, I was thinking if you would generate arbitrary values for testing, you might want to add values on both sides. Usually there's one case for each constructor, but that doesn't apply in this case. So I guess picking one particular view like you did is the best option.

Another option is implementing Unfoldable for FingerTree, Digit and Node.

Do you have a particular use-case in mind?

**Pull request merged**